src/formatting-strings.md

```diff
@@ -0,0 +1,38 @@
+# Format Strings Using `%q`
+
+Whenever formatting messages that contain a string component via `fmt`, use `%q` instead of `%s`. This will wrap the specified string in quotes, helping it stand out from the rest of the error message. More importantly, if the string is empty, it will provide a more helpful error message.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+fmt.Errrof("file %s not found", filename)
+// Prints the following:
+// file myfile.go not found
+//
+// Or if the string is empty:
+// file not found
+```
+
+</td><td>
+
+```go
+fmt.Errrof("file %q not found", filename)
+// Prints the following:
+// file "myfile.go" not found
+//
+// Or if the string is empty:
+// file "" not found
+```
+
+</td></tr>
+</tbody></table>
+
+This advice applies more generally to other contexts when reporting user-specified data, such as logging invalid usernames:
+
+```go
+log.Printf("User %q does not exist", username)
+// User "no_name" does not exist
+```
```

Comment:
I would generalize this slightly to something like:

> This advice can also apply more generally to other formatting situations
> when a dynamic string could confuse or change the formatting or meaning of
> surrounding text when interpolated.

Mostly to avoid bucketing "logging" as "formatting" (i.e., structured logging - which is at least as relevant as `log.Printf` scenarios - doesn't require the same treatment for fields, because they are already discrete values).

I think the `log.Printf` example is fine to keep, I would just put a good/bad example under the `fmt.Errorf` examples.